app/Services/NumberFormatter.php

```diff
@@ -55,6 +55,14 @@ public static function currency($value, string $currency): string
             ->formatCurrency((float) $value, $currency);
     }
 
+    /**
+     * @param string|int|float $value
+     */
+    public static function usdWithDecimals($value, ?int $decimals = 4): string
+    {
+        return '$'.rtrim(rtrim(number_format((float) $value, $decimals ?? 4), '0'), '.');
```

Comment:
On a side note, this might need to still use the NumberFormatter to keep it consistent with the rest of the site, since that formats USD as `US$ 1.12` everywhere else. I think you can do something like the below instead:

```suggestion
    public static function usdWithDecimals($value, ?int $decimals = 4): string
    {
        return rtrim(BetterNumberFormatter::new()
            ->withLocale('international')
            ->withFractionDigits($decimals ?? 4)
            ->formatCurrency((float) $value, 'USD'), '0.');
```

![image](https://github.com/ArdentHQ/arkscan/assets/8069294/a482de39-c77a-481f-a318-137044755e5e)

This will likely get re-reviewed with the other task (https://app.clickup.com/t/862jvjz1w)